src/models/activity.rs

```diff
@@ -284,9 +284,18 @@ impl Activity {
     ) -> Vec<Activity> {
         let mut activities: Vec<Activity> = vec![];
 
-        let compatible_hours_overlay =
-            Self::get_compatible_overlay_hours(goal_to_use, calendar, time_budget);
-
+        let compatible_hours_overlay = Activity::get_compatible_hours_overlay(
```

Comment:
Please undo this code duplication.

Reply:
Actually this is on purpose. See PR #445 for the explanation. I'm actually reducing code duplication. 

I agree more can be done ... but I'll leave that for later. It is not added in this PR. It was already there and can be handled in a separate issue - where we deduplicate at the right level, if this makes sense.

Reply:
There were two functions with nearly identical names before - I've deduplicated this to one.